web/viewer.js

```diff
@@ -297,6 +297,12 @@ var PDFViewerApplication = {
         }
         PDFJS.externalLinkTarget = value;
       }),
+      Preferences.get('withCredentials').then(function resolved(value) {
+        if (PDFJS.withCredentials === true) {
+          return;
+        }
+        PDFJS.withCredentials = value;
```

Comment:
We're planning to remove all global preferences added to PDFJS. They should instead be passed in as arguments (e.g. to open()). It's fine to pass this argument on, but don't use PDFJS to store the value.